### PR TITLE
Index LCC/DDC into solr

### PR DESCRIPTION
**Pull request by cdrini**
Part 3 of #3290 . Updates indexing code to include LCC/DDC

| ~Based on #3302 ; that must be merged first! This is the diff you seek: https://github.com/internetarchive/openlibrary/pull/3338/commits/64d3e3ef848699636ebf03bce70d94d9d90293bf~ |
| -- |

### Technical
- Adds 4 new fields to solr
- Since the new fields haven't been deployed yet, add them behind a config flag (we can't deploy those fields until we have solr-updater running with this new code, (unless we want to do 2 full reindexes (no thank you)).

### Testing
- I tested by adding print statements to the code in question, and confirming that: (1) when flag set to false, the code is not run, (2) when set to true, it is run.
- Tested by running a _full_ solr reindex, and analysing result here: here: https://docs.google.com/spreadsheets/d/1QuZyA6B9nq5mgOc5pmfDLVyLsgX86A_YP75VqYf5mVM
- Try it yourself sample facet query: http://server.openjournal.foundation:8984/solr/select/?q=type:work&version=2.2&start=0&indent=on&rows=0&facet=true&facet.query=lcc:A*&facet.query=lcc:B*&facet.query=lcc:C*&facet.query=lcc:D*&facet.query=lcc:E*&facet.query=lcc:F*&facet.query=lcc:G*&facet.query=lcc:H*&facet.query=lcc:J*&facet.query=lcc:K*&facet.query=lcc:L*&facet.query=lcc:M*&facet.query=lcc:N*&facet.query=lcc:P*&facet.query=lcc:Q*&facet.query=lcc:R*&facet.query=lcc:S*&facet.query=lcc:T*&facet.query=lcc:U*&facet.query=lcc:V*&facet.query=lcc:W*&facet.query=lcc:Z*

### Evidence
No ui changes.

### Stakeholders
@cclauss @mekarpeles @tfmorris

### CODE REVIEW

**Comment by mekarpeles** [comment]
lgtm, @cdrini can you please merge when the dependent issues are landed?

File: openlibrary/solr/update_work.py [hunk 1]
```diff
@@ -510,6 +512,32 @@ def add_list(name, values):
...
+        # Need this to enable in pytest, which have config.runtime_config = {}
+        conf = config.runtime_config or {
+            'plugin_worksearch': {
+                'has_classifications': True
+            }}
+        if conf.get('plugin_worksearch', {}).get('has_classifications', False):
+            raw_lccs = set(lcc
+                           for ed in editions
+                           for lcc in ed.get('lc_classifications', []))
+            lccs = set(lcc for lcc in map(short_lcc_to_sortable_lcc, raw_lccs) if lcc)
```
Comment (cclauss) [inline openlibrary/solr/update_work.py:524]:
```suggestion
            lccs = set(short_lcc_to_sortable_lcc(lcc) for lcc in raw_lccs if lcc)
```

File: openlibrary/tests/solr/test_update_work.py [hunk 2]
```diff
@@ -354,6 +356,74 @@ def test_author_info(self):
         assert d['author_facet'] == ['OL1A Author One', 'OL2A Author Two']
         assert d['author_alternative_name'] == ["Author 1"]
 
+    LCC_TESTS = {
```
Comment (cclauss) [inline openlibrary/tests/solr/test_update_work.py:359]:
```suggestion
    # {'Test name': (doc_lccs, solr_lccs, sort_lcc_index)}
    LCC_TESTS = {
```

File: openlibrary/solr/update_work.py (hunk 1 above)
Comment (cdrini) [reply openlibrary/solr/update_work.py:524]: This slightly different; the result of that function could be `None`, not the input. Marking as won't fix (not an issue).

File: openlibrary/tests/solr/test_update_work.py (hunk 2 above)
Comment (cdrini) [reply openlibrary/tests/solr/test_update_work.py:359]: I've never seen that syntax before; what does this do?

Comment (cdrini) [reply openlibrary/tests/solr/test_update_work.py:359]: Oh, it's just a comment to outline the data's shape :P 👍 Lgtm!

**Pull request merged**